internal/controller/ipaddress_controller.go

```diff
@@ -81,7 +82,14 @@ func (r *IpAddressReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 			if !o.Spec.PreserveInNetbox {
 				err := r.NetboxClient.DeleteIpAddress(o.Status.IpAddressId)
 				if err != nil {
-					return ctrl.Result{}, err
+					if !errors.Is(err, utils.ErrNotFound) {
```

Comment:
if we return nil if we have a 404 error in DeleteIpAdress() then we don't need to check here if it's a not found error.